yearn/apy/curve/simple.py

```diff
@@ -154,80 +165,160 @@ def simple(vault, samples: ApySamples) -> Apy:
 
         performance = (strategist_reward + strategist_performance + treasury) / 1e4
         management = 0
-
-    if isinstance(vault, VaultV2) and len(vault.strategies) == 2:
+    
+    # if the vault consists of only a convex strategy then return 
+    # specialized apy calculations for convex
+    if _ConvexVault.is_convex_vault(vault):
+        cvx_strategy = vault.strategies[0].strategy
+        cvx_vault = _ConvexVault(cvx_strategy, vault, gauge)
+        return cvx_vault.apy(base_asset_price, pool_price, base_apr, pool_apy, management, performance)
+
+    # if the vault has two strategies then the first is curve and the second is convex
+    if isinstance(vault, VaultV2) and len(vault.strategies) == 2: # this vault has curve and convex
         crv_strategy = vault.strategies[0].strategy
         cvx_strategy = vault.strategies[1].strategy
-        convex_voter = addresses[chain.id]['convex_voter_proxy']
-        cvx_working_balance = gauge.working_balances(convex_voter, block_identifier=block)
-        cvx_gauge_balance = gauge.balanceOf(convex_voter, block_identifier=block)
-
-        if cvx_gauge_balance > 0:
-            cvx_boost = cvx_working_balance / (PER_MAX_BOOST * cvx_gauge_balance) or 1
-        else:
-            cvx_boost = MAX_BOOST
-        
-        cvx_booster = contract(addresses[chain.id]['convex_booster'])
-        cvx_lock_incentive = cvx_booster.lockIncentive(block_identifier=block)
-        cvx_staker_incentive = cvx_booster.stakerIncentive(block_identifier=block)
-        cvx_earmark_incentive = cvx_booster.earmarkIncentive(block_identifier=block)
-        cvx_fee = (cvx_lock_incentive + cvx_staker_incentive + cvx_earmark_incentive) / 1e4
-        cvx_keep_crv = cvx_strategy.keepCRV(block_identifier=block) / 1e4
-
-        total_cliff = 1e3
-        max_supply = 1e2 * 1e6 * 1e18 # ?
-        reduction_per_cliff = 1e23
-        cvx = contract(addresses[chain.id]['cvx'])
-        supply = cvx.totalSupply(block_identifier=block)
-        cliff = supply / reduction_per_cliff
-        if supply <= max_supply:
-            reduction = total_cliff - cliff
-            cvx_minted_as_crv = reduction / total_cliff
-            cvx_price = magic.get_price(cvx, block=block)
-            converted_cvx = cvx_price / crv_price
-            cvx_printed_as_crv = cvx_minted_as_crv * converted_cvx
-        else:
-            cvx_printed_as_crv = 0
-
-        cvx_apr = ((1 - cvx_fee) * cvx_boost * base_apr) * (1 + cvx_printed_as_crv) + reward_apr
-        cvx_apr_minus_keep_crv = ((1 - cvx_fee) * cvx_boost * base_apr) * ((1 - cvx_keep_crv) + cvx_printed_as_crv)
-        
+        cvx_vault = _ConvexVault(cvx_strategy, vault, gauge)
         crv_debt_ratio = vault.vault.strategies(crv_strategy)[2] / 1e4
-        cvx_debt_ratio = vault.vault.strategies(cvx_strategy)[2] / 1e4
+        cvx_apy_data = cvx_vault.get_detailed_apy_data(base_asset_price, pool_price, base_apr)
     else:
```

Comment:
I think originally this `else` statement was meant to catch any curve vaults that _only_ had Curve strategies...not sure if we really need it anymore? a bit confused about what you're doing here with it.

Reply:
So it should be exactly the same as before, not sure why I updated `crv_debt_ratio` to 0, it still should be 1. Also not sure that my comment makes sense